File: packages/ocr/android/src/oldarch/OcrSpec.java
package com.ocr;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.Promise;

abstract class OcrSpec extends ReactContextBaseJavaModule {
  OcrSpec(ReactApplicationContext context) {
    super(context);
  }

  public abstract void multiply(double a, double b, Promise promise);
}

File: packages/react-native/android/src/oldarch/OcrSpec.kt
package com.ocr

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule

abstract class RNOcrSpec internal constructor(context: ReactApplicationContext) :
    ReactContextBaseJavaModule(context) {

  // abstract fun create(options: ReadableMap, promise: Promise)

  abstract fun detect(imagePath: String, promise: Promise)

  abstract fun install()
}
